File: Task-03/C/subtask3.c
#include <stdio.h>

int main() {
    int n;

    // Get the number of rows 
    printf("Enter the number of rows for the diamond pattern: ");
    scanf("%d", &n);

    // Upper half
    for (int i = 1; i <= n; i += 2) {
        for (int j = 0; j < (n - i) / 2; j++) printf(" ");
        for (int k = 0; k < i; k++) printf("*");
        printf("\n");
    }

    // Lower half
    for (int i = n - 2; i >= 1; i -= 2) {
        for (int j = 0; j < (n - i) / 2; j++) printf(" ");
        for (int k = 0; k < i; k++) printf("*");
        printf("\n");
    }

    return 0;
}

File: Task-03/Go/subtask3.go
package main

import "fmt"

func main() {
	var n int
	fmt.Print("Enter a number: ")
	fmt.Scan(&n)

	// Upper half of the diamond
	for i := 1; i <= n; i += 2 {
		for j := 0; j < (n-i)/2; j++ {
			fmt.Print(" ")
		}
		for k := 0; k < i; k++ {
			fmt.Print("*")
		}
		fmt.Println()
	}

	for i := n - 2; i > 0; i -= 2 {
		for j := 0; j < (n-i)/2; j++ {
			fmt.Print(" ")
		}
		for k := 0; k < i; k++ {
			fmt.Print("*")
		}
		fmt.Println()
	}
}
